wal/wal.go

```diff
@@ -617,15 +621,21 @@ func (w *WAL) log(rec []byte, final bool) error {
 
 		// By definition when a record is split it means its size is bigger than
 		// the page boundary so the current page would be full and needs to be flushed.
-		// On contrary if we wrote a full record, we can fit more records of the batch
-		// into the page before flushing it.
-		if final || typ != recFull || w.page.full() {
+		if typ == recFirst || typ == recMiddle || w.page.full() {
```

Comment:
Looking at the above comment, I was thinking whether checking just `w.page.full()` would be enough here.

Reply:
@codesome Totally agree with you, updated.